src/graphics/entity.rs

```diff
@@ -0,0 +1,91 @@
+use glium::implement_vertex;
+
+use crate::vector::Vector3;
+
+pub const ENTITY_VERTEX_SHADER: &str = r#"
+        #version 150
+
+        in vec3 position;
+        in mat4 world_matrix;
+
+        // The vertex shader has some passthrough for the fragment shader...
+
+        // Which face of the cube is being passed ?
+        in int face;
+        flat out int face_s;
+
+        // Index of the block to be used
+        in int body_part_id;
+        flat out int body_part_id_s;
+
+        // Where is the vertex located on the face ?
+        in vec2 tex_coords;
+        out vec2 v_tex_coords;
+
+        uniform mat4 perspective;
+        uniform mat4 view;
+
+        void main() {
+            gl_Position = perspective * view * world_matrix * vec4(position, 1.0);
+            v_tex_coords = tex_coords;
+            face_s = face;
+            body_part_id_s = body_part_id;
+        }
+    "#;
+
+// Fragment shader
+pub const ENTITY_FRAGMENT_SHADER: &str = r#"
+        #version 140
+
+        // passed-through the vertex shader
+        flat in int face_s;
+        flat in int body_part_id_s;
+        in vec2 v_tex_coords;
+
+        out vec4 color ;
+
+        uniform sampler2DArray textures;
+        
+        void main() {
+            // Each block has 6 types of faces
+            int idx = body_part_id_s * 6;
+            color = texture(textures, vec3(v_tex_coords, idx + face_s));
+         }
+    "#;
+
+/// An OpenGL type that contains the information for OpenGL's instancing
+#[derive(Copy, Clone)]
+pub struct EntityCube {
+    world_matrix: [[f32; 4]; 4],
+    body_part_id: u8,
+}
+
+implement_vertex!(EntityCube, world_matrix, body_part_id);
+
+impl EntityCube {
+    /// Build a rendered cube center around position (and not around position + (0.5,0.5,0.5) as for CubeAttr !!!!)
+    /// The cube is scaled by scale for each direction
+    /// A Yaw rotation is applied (first component of rot)
+    /// A Pitch roation is then apllied (second component of rot)
+    // Maybe implement roll one day ?
+    pub fn new(position: &Vector3, body_part_id: u8, scale: [f32; 3], rot: [f32; 2]) -> Self {
```

Comment:
Considering the usage of this function (1)

(1) in _humanoid::draw_ (to be renamed by the way) the function is mostly called with _rot = [x, 0.]_, I recommend to instead of using an array of size two, it's better to pass two floats: _yaw: f32, pitch: f32_. This will avoid unnecessary allocations.